docker/server-jetty/build.gradle

```diff
@@ -0,0 +1,76 @@
+import io.deephaven.tools.docker.Architecture
+import io.deephaven.tools.License
+
+plugins {
+    id 'com.bmuschko.docker-remote-api'
+    id 'io.deephaven.project.register'
+}
+
+def targetArch = Architecture.targetArchitecture(project)
+
+def baseMapAmd64 = [
+        'server-base': 'server-jetty',
+]
+
+// Only the server image is supported on arm64
+// TODO(deephaven-core#1702): Support arm64 builds for applicable extended images
+def baseMapArm64 = baseMapAmd64.findAll { key, value -> value == 'server-jetty' }
+
+def baseMap = targetArch == Architecture.AMD64 ? baseMapAmd64 : baseMapArm64
+
+baseMap.keySet().each {base ->
+    evaluationDependsOn Docker.registryProject(base)
+}
+
+configurations {
+    serverApplicationDist
+    pythonWheel
+}
+
+dependencies {
+    serverApplicationDist project(path: ':server-jetty-app', configuration: 'applicationDist')
+
+    pythonWheel project(':py-server')
+}
+
+def dockerLicenses = License.createFrom(project).syncDockerLicense().get()
+
+def context= project.layout.buildDirectory.dir('context')
+
+def prepareDocker = project.tasks.register('prepareDocker', Sync) {
+
+    from 'src/main/docker'
+    baseMap.values().each { server ->
+        from("src/main/${server}") {
+            into server
+        }
+    }
+    from ('src/main/configure') {
+        into 'configure'
+    }
+
+    from(configurations.pythonWheel) {
+        into 'wheels'
+    }
+
+    from(configurations.serverApplicationDist) {
+        into 'serverApplicationDist'
+    }
+    from(dockerLicenses.outputs) {
+        into 'licenses'
+    }
+    into context
+}
+
+def buildDockerClosure = { String base, String server ->
```

Comment:
The netty build has:
```
// TODO(deephaven-core#1596): Remove extra dependencies for build-ci.yml
// Currently, GH build-ci.yml calls prepareDockerAll, and relies on buildx; so we need to
// make sure that dockerCreateDockerfile has the proper dependencies.
// If we change how our build works in the future (so that gradle is responsible for pushing the
// image), we can remove this extra dependency.
project.tasks.register('prepareDockerAll') {
    dependsOn prepareDocker, baseMap.keySet().collect { base -> Docker.registryTask(project, base) }
}
```

Why don't we need this, just because we're still using netty for CI?

Reply:
This was mostly a copy-paste from the netty logic; I'm not sure if this specific block is still necessary since we don't have an CI artifacts for docker + jetty in deephaven-core.

There are non-trivial dependencies beyond for other build related things that depend on server-netty. For example, sphinx. I'm betting these will be "easy" to fix over to server-jetty; but it doesn't serve the purpose of the release IMO so I tried to limit the scope as much as possible.

Technically, there are a bunch of folder renaming that I didn't need to do - but I did anyways, just to keep things clearer w/ the universal -netty suffix.

The build logic in server-jetty is meant to serve local development purposes for `./gradlew prepareCompose; docker compose up -d` workflows.